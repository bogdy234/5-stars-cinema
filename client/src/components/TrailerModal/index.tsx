import { FC, ReactElement, useRef } from "react";
import useClickOutside from "../../utils/hooks/useClickOutside";
import Modal from "../Modal";

interface TrailerModalProps {
    showModal: boolean;
    youtubeLink: string;
    toggle: () => void;
}

const TrailerModal: FC<TrailerModalProps> = ({
    showModal,
    youtubeLink,
    toggle,
}): ReactElement => {
    const ref = useRef(null);
    useClickOutside(ref, toggle);
    return (
        <Modal showModal={showModal}>
            <div className="h-screen flex justify-center items-center">
                <iframe
                    ref={ref}
                    width="853"
                    height="480"
                    src={`https://www.youtube.com/embed/${youtubeLink.substring(
                        youtubeLink.indexOf("watch?v=") + "watch?v=".length,
                        youtubeLink.length
                    )}`}
                    frameBorder="0"
                    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                    allowFullScreen
                    title="Trailer"
                />
            </div>
        </Modal>
    );
};

export default TrailerModal;
