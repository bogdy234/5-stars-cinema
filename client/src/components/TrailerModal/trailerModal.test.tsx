import renderer from "react-test-renderer";
import TrailerModal from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <TrailerModal showModal={false} youtubeLink={""} toggle={jest.fn()} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
