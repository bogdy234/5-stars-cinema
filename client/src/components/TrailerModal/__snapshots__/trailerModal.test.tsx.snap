// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed z-10 left-0 top-0 w-full h-full overflow-auto bg-black bg-opacity-50 hidden"
>
  <div
    className="h-screen flex justify-center items-center"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      frameBorder="0"
      height="480"
      src="https://www.youtube.com/embed/"
      title="Trailer"
      width="853"
    />
  </div>
</div>
`;
